server/game/cards/01_SOR/events/DontGetCocky.ts

```diff
@@ -0,0 +1,74 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { AbilityContext } from '../../../core/ability/AbilityContext';
+import { EventCard } from '../../../core/card/EventCard';
+import { GameStateChangeRequired, TargetMode, WildcardCardType } from '../../../core/Constants';
+import { GameSystem } from '../../../core/gameSystem/GameSystem';
+import { IThenAbilityPropsWithSystems } from '../../../Interfaces';
+import * as Contract from '../../../core/utils/Contract';
+
+export default class DontGetCocky extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '2202839291',
+            internalName: 'dont-get-cocky',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Choose a unit',
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.Unit,
+                mustChangeGameState: GameStateChangeRequired.None
+            },
+            then: (unitChosenContext) => ({
+                title: 'Reveal the top card of your deck',
+                immediateEffect: AbilityHelper.immediateEffects.reveal((context) => ({ target: context.source.controller.getTopCardOfDeck() })),
+                then: this.thenAfterReveal(1, unitChosenContext)
+            })
+        });
+    }
+
+    private thenAfterReveal(cardsRevealedCount: number, contextWithUnitTarget: AbilityContext): IThenAbilityPropsWithSystems<AbilityContext> {
+        Contract.assertTrue(cardsRevealedCount > 0 && cardsRevealedCount < 8, `Error in Don\'t Get Cocky implementation: thenAfterReveal called with invalid cardsRevealedCount ${cardsRevealedCount}`);
+        if (cardsRevealedCount < 7) {
+            const then: IThenAbilityPropsWithSystems<AbilityContext> = {
+                title: 'Reveal the next card from your deck or stop revealing cards',
+                targetResolver: {
+                    mode: TargetMode.Select,
+                    choices: {
```

Comment:
I was thinking it might be good to set activePromptTitle to show the total cost so far. Maybe just something like `Current total cost: ${totalCost}\nSelect one:`. We would want to have a check in the tests for that as well